host/allwinner/gpio.go

```diff
@@ -366,50 +375,11 @@ func (p *Pin) Out(l gpio.Level) error {
 // to reach Mhz scale bit banging.
 func (p *Pin) FastOut(l gpio.Level) {
 	bit := uint32(1 << p.offset)
-	// Pn_DAT  n*0x24+0x10  Port n Data Register (n from 1(B) to 7(H))
-	switch p.group {
-	case 1:
-		if l {
-			drvGPIO.gpioMemory.groups[1].data |= bit
-		} else {
-			drvGPIO.gpioMemory.groups[1].data &^= bit
-		}
-	case 2:
-		if l {
-			drvGPIO.gpioMemory.groups[2].data |= bit
-		} else {
-			drvGPIO.gpioMemory.groups[2].data &^= bit
-		}
-	case 3:
-		if l {
-			drvGPIO.gpioMemory.groups[3].data |= bit
-		} else {
-			drvGPIO.gpioMemory.groups[3].data &^= bit
-		}
-	case 4:
-		if l {
-			drvGPIO.gpioMemory.groups[4].data |= bit
-		} else {
-			drvGPIO.gpioMemory.groups[4].data &^= bit
-		}
-	case 5:
-		if l {
-			drvGPIO.gpioMemory.groups[5].data |= bit
-		} else {
-			drvGPIO.gpioMemory.groups[5].data &^= bit
-		}
-	case 6:
-		if l {
-			drvGPIO.gpioMemory.groups[6].data |= bit
-		} else {
-			drvGPIO.gpioMemory.groups[6].data &^= bit
-		}
-	case 7:
-		if l {
-			drvGPIO.gpioMemory.groups[7].data |= bit
-		} else {
-			drvGPIO.gpioMemory.groups[7].data &^= bit
-		}
+	// Pn_DAT  n*0x24+0x10  Port n Data Register (n from 0(A) to 8(I))
```

Comment:
This was found to be significantly slower. IIRC it's because of the boundary checks. Please revert to the expanded case.
There's notes at https://periph.io/news/2017/gpio_perf/.

Reply:
Oh that's really interesting, I knew there must've been a reason :)
I've changed it back and added a comment pointing to the article.